crates/fuzzy/src/strings.rs

```diff
@@ -57,10 +57,10 @@ pub struct StringMatch {
 }
 
 impl StringMatch {
-    pub fn ranges<'a>(&'a self) -> impl 'a + Iterator<Item = Range<usize>> {
+    pub fn ranges(&self) -> impl '_ + Iterator<Item = Range<usize>> {
         let mut positions = self.positions.iter().peekable();
         iter::from_fn(move || {
-            while let Some(start) = positions.next().copied() {
+            if let Some(start) = positions.next().copied() {
```

Comment:
The `return Some(start..end);` below meant that we would never do more than one iteration of this outer `while` loop.

This was flagged by Clippy's [`never_loop`](https://rust-lang.github.io/rust-clippy/master/index.html#/never_loop) rule.